guppylang/std/qsystem/__init__.py

```diff
@@ -74,6 +76,42 @@ def reset(q: qubit) -> None: ...
 def qfree(q: qubit @ owned) -> None: ...
 
 
+@guppy.hugr_op(quantum_op("LazyMeasureLeaked", ext=QSYSTEM_EXTENSION))
+@no_type_check
+def _measure_leaked(q: qubit @ owned) -> Future[int]:
+    """Measure the quibit or return 2 if it is leaked."""
+
+
+@guppy
+def measure_leaked(q: qubit @ owned) -> "MaybeLeaked":
+    fm = _measure_leaked(q)
+    return MaybeLeaked(fm)
+
+@guppy
+class MaybeLeaked:
+    """A class representing a measurement that may have leaked.
+
+    This is used to represent the result of `measure_leaked`, which can either
+    return a boolean measurement result or indicate that the qubit has leaked.
+    """
+
+    _measurement: Future[int]
+
+    def __init__(self, value: Future[int]):
+        self._measurement = value
+
+    def is_leaked(self) -> bool:
```

Comment:
Methods must be annotated as Guppy and `self` needs a type annotation:

```suggestion
    def is_leaked(self: "MaybeLeaked @ owned") -> bool:
```